daemon/daemon.go

```diff
@@ -1501,6 +1501,16 @@ func (daemon *Daemon) newBaseContainer(id string) *container.Container {
 	return container.NewBaseContainer(id, daemon.containerRoot(id))
 }
 
+// Reload reads configuration changes and modifies the
+// daemon according to those changes.
+// Currently, only the labels configuration is taken into account.
```

Comment:
Well, labels can easily be changed ad-hoc.  Going from no discovery config to an initial config is super beneficial when bootstrapping a new node in a production environment.  Deferring live reconfiguration of the discovery setup seems reasonable to me (one can argue that the operation is by nature disruptive, so requiring a restart actually makes sense.)